ui/StatusQ/src/leftjoinmodel.cpp

```diff
@@ -227,18 +321,18 @@ void LeftJoinModel::setJoinRole(const QString& joinRole)
 
     emit joinRoleChanged();
 
-    initializeIfReady();
+    initializeIfReady(true);
 }
 
 const QString& LeftJoinModel::joinRole() const
 {
     return m_joinRole;
 }
 
-void LeftJoinModel::setSourceModel(QAbstractItemModel* newSourceModel)
+int LeftJoinModel::rowCount(const QModelIndex &parent) const
 {
-    qWarning() << "Source model is not intended to be set directly on this model."
-                  " Use setLeftModel and setRightModel instead!";
+    return m_leftModel == nullptr || !m_initialized
```

Comment:
This will fix my tests. The model tester thinks it's dealing with a tree, not a list.
```suggestion
    if (parent.isValid())
        return 0;

    return m_leftModel == nullptr || !m_initialized
```